lib/_tls_wrap.js

```diff
@@ -846,10 +846,23 @@ Server.prototype.setOptions = function(options) {
     this.rejectUnauthorized = false;
   }
 
+
   if (options.pfx) this.pfx = options.pfx;
-  if (options.key) this.key = options.key;
+
+  if (!options.key) {
+    throw new Error('key is a required parameter for Server.createServer');
+  } else {
+    this.key = options.key;
+  }
+
   if (options.passphrase) this.passphrase = options.passphrase;
-  if (options.cert) this.cert = options.cert;
+
+  if (!options.cert) {
```

Comment:
Right now I test this way: 
`assert.throws(() => https.createServer(options1, assert.fail),'cert is a required parameter for server.createServer');`
Is one better than the other? If so I am curious to understand why?